src/Features/CSharp/Portable/Diagnostics/Analyzers/TypeSyntaxSimplifierWalker.cs

```diff
@@ -200,6 +242,11 @@ public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax no
 
         public override void VisitQualifiedCref(QualifiedCrefSyntax node)
         {
+            if (_ignoredSpans?.HasIntervalThatOverlapsWith(node.FullSpan.Start, node.FullSpan.Length) ?? false)
+            {
+                return;
+            }
```

Comment:
would it potentially make more sense to do htis in some sort of DefaultVisit method?  i.e. don't bother descending into things once it has a span that is fully in an inored span?

Also, it seems like a big win to disable the moment you hit a `block` or `arrow expression`.  that way you can get out quickly instead of still walking *all* the nodes of these and having to do the HasIntervalThatOverlapsWith for every single matching node underneath these nodes.

Reply:
`HasIntervalThatOverlapsWith` isn't especially cheap. It pays off to call it to avoid the more expensive `TrySimplify`, but otherwise it would be good to only call it when it's likely to avoid substantially more work. `VisitBlock` (or `VisitMethodDefinition`) is a candidate.

Reply:
I'm going to leave this one for now. The current approach doesn't leave measurable overhead in the walker that needs to be addressed, so we can keep the simplicity of only checking the interval tree in order to avoid a call to `TrySimplify`.